src/v2i-hub/CARMAStreetsPlugin/src/CARMAStreetsPlugin.cpp

```diff
@@ -64,96 +67,44 @@ void CARMAStreetsPlugin::UpdateConfigSettings() {
-		PLOG(logERROR) <<"Setting kafka config options failed with error:" << error_string << "\n" <<"Exiting with exit code 1";
-		exit(1);
-	} else {
-		PLOG(logDEBUG) <<"Kafka config options set successfully";
-	}
-	
-	kafka_producer = RdKafka::Producer::create(kafka_conf, error_string);
-	if (!kafka_producer) {
-		PLOG(logERROR) <<"Creating kafka producer failed with error:" << error_string << "\n" <<"Exiting with exit code 1";
-		exit(1);
-	} 			
-	PLOG(logDEBUG) <<"Kafka producer created";
-
-	if (kafka_conf_sp_consumer->set("bootstrap.servers", kafkaConnectString, error_string)  != RdKafka::Conf::CONF_OK
-		 || (kafka_conf_sp_consumer->set("group.id", _subscribeToSchedulingPlanConsumerGroupId, error_string) != RdKafka::Conf::CONF_OK)
-		 || (kafka_conf_spat_consumer->set("bootstrap.servers", kafkaConnectString, error_string)  != RdKafka::Conf::CONF_OK)
-		 || (kafka_conf_spat_consumer->set("group.id", _subscribeToSpatConsumerGroupId, error_string) != RdKafka::Conf::CONF_OK)
-		 || (kafka_conf_ssm_consumer->set("bootstrap.servers", kafkaConnectString, error_string)  != RdKafka::Conf::CONF_OK)
-		 || (kafka_conf_ssm_consumer->set("group.id", _subscribeToSSMConsumerGroupId, error_string) != RdKafka::Conf::CONF_OK)
-		 ) {
-		PLOG(logERROR) <<"Setting kafka config group.id options failed with error:" << error_string << "\n" <<"Exiting with exit code 1";
-		exit(1);
-	} else {
-		PLOG(logDEBUG) <<"Kafka config group.id options set successfully";
-	}
-	kafka_conf_sp_consumer->set("enable.partition.eof", "true", error_string);
-	kafka_conf_spat_consumer->set("enable.partition.eof", "true", error_string);
-	kafka_conf_ssm_consumer->set("enable.partition.eof", "true", error_string);
-
-	_scheduing_plan_kafka_consumer = RdKafka::KafkaConsumer::create(kafka_conf_sp_consumer, error_string);
-	_spat_kafka_consumer = RdKafka::KafkaConsumer::create(kafka_conf_spat_consumer, error_string);
-	_ssm_kafka_consumer = RdKafka::KafkaConsumer::create(kafka_conf_ssm_consumer, error_string);
-
-	if ( !_scheduing_plan_kafka_consumer || !_spat_kafka_consumer || !_ssm_kafka_consumer) {
-		PLOG(logERROR) << "Failed to create Kafka consumers: " << error_string << std::endl;
-		exit(1);
-	}
-	PLOG(logDEBUG) << "Created consumer " << _scheduing_plan_kafka_consumer->name() << std::endl;
-	PLOG(logDEBUG) << "Created consumer " << _spat_kafka_consumer->name() << std::endl;
-	PLOG(logDEBUG) << "Created consumer " << _ssm_kafka_consumer->name() << std::endl;
-
-	//create kafka topics
-	RdKafka::Conf *tconf_spat = RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC);
-	RdKafka::Conf *tconf_sp = RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC);
-	RdKafka::Conf *tconf_ssm = RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC);
-	if(!tconf_spat && !tconf_sp)
+	kafka_client client;
+
+	//Producer
+	_kafka_producer_ptr = client.create_producer(kafkaConnectString);
+
+	//Consumers
+	auto uuid = boost::uuids::random_generator()();
```

Comment:
We talked about removing this. There is no reason to have a configuration consumer group id if it is appended with random strings.

Reply:
I think this UUID is the temporary fix to prevent two consumers from joining the same group causing rebalancing.

Reply:
@dan-du-car can we remove this from the PR description as well so the release notes accurately reflect changes.

Reply:
Done